src/Common/src/Interop/Linux/procfs/Interop.ProcFsStat.cs

```diff
@@ -285,5 +302,73 @@ internal static bool TryParseStatFile(string statFilePath, out ParsedStat result
             result = results;
             return true;
         }
+
+        internal static bool TryReadStatusFile(int pid, out ParsedStatus result, ReusableTextReader reusableReader)
+        {
+            bool b = TryParseStatusFile(GetStatusFilePathForProcess(pid), out result, reusableReader);
+            Debug.Assert(!b || result.pid == pid, "Expected process ID from status file to match supplied pid");
+            return b;
+        }
+
+        private static bool TryParseStatusFile(string statusFilePath, out ParsedStatus result, ReusableTextReader reusableReader)
+        {
+            string statusFileContents;
```

Comment:
The next few lines are duplicated in TryReadStatFile etc, you could extract them into something like ReadFileContentNoThrow that returned a string or null.